### PR TITLE
添加nrf5x的gpio驱动

### PR DESCRIPTION
**Pull request by guohp1128**
添加和修改6个文件

### CODE REVIEW

File: bsp/nrf5x/libraries/drivers/drv_gpio.c [hunk 1]
```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (c) 2006-2018, RT-Thread Development Team
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author            Notes
+ * 2018-11-06     balanceTWK        first version
```
Comment (supperthomas) [inline bsp/nrf5x/libraries/drivers/drv_gpio.c:8]: please fixed by your name

File: bsp/nrf5x/nrf52840/board/sdk_config.h [hunk 2]
```diff
@@ -3556,27 +3556,27 @@
 // <e> NRFX_SPI_ENABLED - nrfx_spi - SPI peripheral driver
 //==========================================================
 #ifndef NRFX_SPI_ENABLED
-#define NRFX_SPI_ENABLED 1
+#define NRFX_SPI_ENABLED 0
```
Comment (supperthomas) [inline bsp/nrf5x/nrf52840/board/sdk_config.h:3559]: please open the SPI enable

File: bsp/nrf5x/nrf52840/board/sdk_config.h [hunk 3]
```diff
@@ -11694,11 +11694,9 @@
 #endif
 // </e>
 
-// </h>
+// </h> 
 //==========================================================
-#ifndef NRFX_SYSTICK_ENABLED
-#define NRFX_SYSTICK_ENABLED 1
-#endif
+
```
Comment (supperthomas) [inline bsp/nrf5x/nrf52840/board/sdk_config.h:11699]: if not needed， no need chang the repo type

File: bsp/nrf5x/nrf52840/applications/application.c [hunk 4]
```diff
@@ -25,24 +25,154 @@
 #include <shell.h>
 #endif
 
-#include <nrf_gpio.h>
+#include <drv_gpio.h>
```
Comment (supperthomas) [inline bsp/nrf5x/nrf52840/applications/application.c:28]: please use the original application
just change the
api nrf_gpio_pin_set  to rt_pin_write。
you can see the file on bsp\stm32\stm32f072-st-nucleo\applications\main.c
just the same。
the gpio test code，is no needed，just for test by yourself。

File: bsp/nrf5x/libraries/drivers/drv_gpio.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2006-2018, RT-Thread Development Team
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author            Notes
+ * 2018-11-06     balanceTWK        first version
```
Comment (supperthomas) [inline bsp/nrf5x/libraries/drivers/drv_gpio.h:8]: fixed by your name

**Comment by BernardXiong** [comment]
Please sign CLA.

**Comment by guohp1128** [comment]
不提交了

**Pull request closed without merging**